Barebones documentation for math types

aumetra opened the pull request.

webmaster128 commented on src/pages/core/standard-library/math/decimals.mdx:
```diff
@@ -0,0 +1,24 @@
+These decimal types come in two variants:
+
+- `Decimal`: 128-bit
+- `Decimal256`: 256-bit
```
- Here it might be worth showing the value range as this is not exactly intuitive given the 18 decimal places
- We can add the signed decimal types as well to this page

webmaster128 commented on src/pages/core/standard-library/math/integers.mdx:
```diff
@@ -0,0 +1,50 @@
+Reasoning behind wrapping the native 64 bit type is the interaction with
+JavaScript. A `u64` would normally serialize into a regular integer field in
+JSON. The issue here is that, due to representing integers in form of double
+precision floats, JavaScript can only handle up to ~48 bit numbers without
```
```suggestion
precision floats, JavaScript can only handle up to ~53 bit numbers without
```

This is how I always got it:

```
$ node
> Math.log(Number.MAX_SAFE_INTEGER) / Math.log(2)
53
```

aumetra replied:
🤦‍♀️ ugh.. I was mixing the bits up with pointers on AMD64. Thanks

webmaster128 commented on src/pages/core/standard-library/math/integers.mdx:
```diff
@@ -0,0 +1,50 @@
+                            <td>{item} bit</td>
+                            <td><code>Uint{item}</code></td>
+                            <td><code>Int{item}</code></td>
+                        </tr>
```
Is that a good place to link to docs.rs for each type?

webmaster128 commented on src/pages/core/standard-library/math/integers.mdx:
```diff
@@ -0,0 +1,50 @@
+512 bits.
+
+Reasoning behind wrapping the native 64 bit type is the interaction with
+JavaScript. A `u64` would normally serialize into a regular integer field in
```
This is not only JavaScript specific. Other JSON parsets like jq behave the same (all numbers are stired as double). That being said, JavaScript is certainly the most important for us.

webmaster128 reviewed (commented):
🙌

hashedone reviewed (commented):
Some comments left in vercel.

aumetra commented on src/lib/_decimals.js:
```diff
@@ -0,0 +1,18 @@
+function prepareName(isSigned, bits) {
+  let name = "Decimal";
+  if (bits != "128") {
```
Maybe we'll manage during the integer/decimal redesign :P

aumetra commented:
Gonna go ahead and merge it as is as I addressed all the review comments for now.

@hashedone If you have more comments, feel free to add them! I'll then address them in a separate PR
